libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -975,6 +967,50 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 
 		break;
 	}
+	case FunctionType::Kind::ECRecover:
+	case FunctionType::Kind::RIPEMD160:
+	case FunctionType::Kind::SHA256:
+	{
+		static map<FunctionType::Kind, std::tuple<u160, size_t>> precompiles = {
+			{FunctionType::Kind::ECRecover, std::make_tuple(1, 12)},
+			{FunctionType::Kind::SHA256, std::make_tuple(2, 0)},
+			{FunctionType::Kind::RIPEMD160, std::make_tuple(3, 12)},
+		};
+		auto [ address, offset ] = precompiles[functionType->kind()];
+		TypePointers argumentTypes;
+		vector<string> argumentStrings;
+		for (auto const& arg: arguments)
+		{
+			argumentTypes.emplace_back(&type(*arg));
+			if (IRVariable(*arg).type().sizeOnStack() > 0)
+				argumentStrings.emplace_back(IRVariable(*arg).commaSeparatedList());
+		}
+		string argumentString = argumentStrings.empty() ? ""s : (", " + joinHumanReadable(argumentStrings));
+		ABIFunctions abi(m_context.evmVersion(), m_context.revertStrings(), m_context.functionCollector());
+		Whiskers templ(R"(
+			let <pos> := <freeMemory>
+			let <end> := <encodeArgs>(<pos> <argumentString>)
+			let <success> := <call>(gas(), <address> <?isCall>, 0</isCall>, <pos>, sub(<end>, <pos>), 0, 32)
+			let <retVars>
+			if <success> {
```

Comment:
Shouldn't it revert on failure?

Reply:
i revert now for `ripemd160` and `sha256`, but not for `ecrecover`.

Reply:
In the old codegen, it reverted for all of them.

Reply:
Ah ok, then I didn't understand you correctly. I thought you said that `ecrecover` will never revert, but always return the result-code.